gmime/gmime-signature.c

```diff
@@ -257,6 +257,49 @@ g_mime_signature_get_expires (GMimeSignature *sig)
 }
 
 
+/**
+ * g_mime_signature_get_created_usecs:
+ * @sig: a #GMimeSignature
+ * @created: (out) pointer to output creation time [out]
+ *
+ * Get the creation date of the signature in microseconds since the UNIX epoch.
+ *
+ * Returns: TRUE if successful; FALSE if @esig is not a signature, @created is NULL, or there is no known creation date for the signature.
+ **/
+gboolean
+g_mime_signature_get_created_usecs (const GMimeSignature *sig, guint64 *created)
+{
+	g_return_val_if_fail (GMIME_IS_SIGNATURE (sig), FALSE);
+	g_return_val_if_fail (created != NULL, FALSE);
+	g_return_val_if_fail (sig->created != (time_t)-1, FALSE);
+
```

Comment:
g_return macros shouldn't be used for valid values and -1 is a valid value here.

Reply:
-1 is not a valid known value -- it's a value meaning "unknown", which this proposed alternate interface represents instead with a return code, instead of the output.